service/controller/v14/adapter/adapter.go

```diff
@@ -73,41 +44,30 @@ type Config struct {
 	StackState        StackState
 }
 
-func NewGuest(cfg Config) (Adapter, error) {
-	a := Adapter{
-		Instance:       &instanceAdapter{},
-		LifecycleHooks: &lifecycleHooksAdapter{},
-		Outputs:        &outputsAdapter{},
-	}
-
-	a.ASGType = prefixWorker
-	a.ClusterID = key.ClusterID(cfg.CustomObject)
-	a.WorkerImageID = cfg.StackState.WorkerImageID
-	// set api whitelisting
-	a.APIWhitelistEnabled = cfg.APIWhitelist.Enabled
+type Adapter struct {
+	Guest    guestAdapter
```

Comment:
Having public properties defined by private types does not make sense to me. So either all public or all private, but the mix does not make sense in my head.

Reply:
We can't go all private because they won't be accessible in templates. In this case I don't have strong opinion. I just made them public. But I wonder why this is allowed by the compiler.